### ISSUE TITLE
Cypress Server Routes Are Lost After Page Load If Not Using cy.visit()

### ISSUE THREAD

**Issue by pihish** [cypress-io/cypress#177]
When I have XHR requests go out on initial page load and need to have Cypress watch for those requests, I have to use `cy.visit()` to get to the page I want to test.  I cannot get directed to that page or else the routes I tell Cypress about are lost.

Example:

``` javascript
cy.visit('https://mydomain.com');
// I am looking for these two XHR requests that are immediately started after my page loads
cy.server().route('GET', /getUsers/).as('getUsers')
           .route('GET', /getApples/).as('getApples');
// By clicking on an element with href value of "/apples", I get directed to a new page
// Cypress will lose the routes I've defined just before this because I've loaded a whole new page
cy.get('#links [href="/apples"]').click().wait(['@getUsers', '@getApples']);
```

We need a feature where even if a new page loads, the routes you define are not lost.  This gives testing a more "natural" flow.  `cy.visit()` is the equivalent of typing in a browser URL.  We should be testing how users actually get from page to page, which is usually clicking on a button, or an image, etc...

**Comment by brian-mann** [comment]
We need to rewrite the XHR proxy handling to be outside of the browser, which we'll be able to do sometime in `0.17.x`

**Comment by brian-mann** [comment]
The code for this is done, but this has yet to be released. We'll update the issue and reference the changelog when it's released.

**Comment by felippepuhle** [comment]
Hi there! Is this still happening? 🤔

**Comment by jennifer-shehane** [comment]
@felippepuhle This should have been fixed in 0.20.0, are you experiencing an issue?

**Comment by felippepuhle** [comment]
@jennifer-shehane nope! I can't remember what I was doing wrong, but it's 100% now. Sorry didn't answer before, and thanks!
